### ISSUE TITLE
Popover+ ShellBarItem: Unable to open Popover with "opener" and "open"

### ISSUE THREAD

**Issue by C3ntraX** [SAP/ui5-webcomponents#5344]
### **Bug Description**
Popover supports the new properties "opener" and "open" but they dont work if you want the popover to open with the new "opener" property with ShellBarItem as target. See codeexample

### **Expected Behavior**
Define a opernerId for ShellBarItem and be able to open the popover with the defined id.

_With this way to open a popover you are able to lazy load the component_, otherwise not (or I don´t know how).

### **Steps to Reproduce**
1. Click on the icon in the Shellbar
2. Nothing happens

### **Isolated Example**
https://codesandbox.io/s/shellbaritem-popover-lj45b1?file=/src/App.js

### **Context**
- UI5 Web Components version: {...}
- OS/Platform: {...}
- Browser: {...}
- Affected component: {...}

### **Log Output / Stack Trace / Screenshots**
```
{...}
```
### **Priority**
- [ ] Low
- [x] Medium
- [ ] High
- [ ] Very High

The priority indicates the severity of the issue. To set the appropriate priority consider the following criteria:
- Breaks entire application or system - High or Very High
- Accessibility issue - Medium or High
- Functional issue - Medium or High
- Visual issue - Low or Medium

**Note:** The priority might be re-evaluated by the issue processor.

### **Stakeholder Info** (if applicable)

- Organization: {...}
- Business impact: {...}

**Comment by ilhan007** [comment]
Hello @C3ntraX 

There are two problems...

1. one on your side - the target reference/opener is not correct.


If you go to the Shellbar samples https://sap.github.io/ui5-webcomponents/main/playground/components/ShellBar/
you will find code such as

```js
shellbar.addEventListener("profile-click", function(event) {
		popover.showAt(event.detail.targetRef);
	});
```
The code still uses the old API via the method, but shows how to provide the correct opener.
The `event.detail.targetRef` is the actual rendered DOM element, that users click as the ShellBarItem is an abstract element that does not have a DOM on its own. That's why you need to provide its matching DOM element ID using the event detail and  `event.detail.targetRef.id`.

So in your case:
```js
function App() {
  const [open, setOpen] = useState(false);
  const [opener, setOpener] = useState("");

  return (
    <>
      <ShellBar>
        <ShellBarItem
          id="notificationOpener"
          onClick={(e) => {
            setOpen(true);
            setOpener(e.detail.targetRef.id);
          }}
          icon="bell"
        />
      </ShellBar>
      <Popover open={open} opener={opener}>
        Some text.......
      </Popover>
    </>
  );
}

```

Now everything is correct but then comes problem number two

2. The Popover does not open as the opener is in another ShadowDom (which means another document) while the Popover gets it  via "const opener = document.getElementById(this.opener);" and expects to find it in the same document.

**Comment by ilhan007** [comment]
Hello @SAP/ui5-webcomponents-topic-rd 

@C3ntraX is trying to open a Popover by using the declarative API (open, opener). Unfortunately, the opener is inside the ShellBar's Shadow DOM and the Popover can't get it via `const opener = document.getElementById(this.opener);`

We need to either think of an improvement or document this as limitation.

P.S. This is an updated  sample https://codesandbox.io/s/shellbaritem-popover-forked-zdiqve based on the author's one, but with correctly set opener ID, so you better use this one for testing.

**Comment by C3ntraX** [comment]
The big opportunity is to lazy load the component with this approach.

Like: 

```
{
  open && <Popover open={open} opener={opener}>
          Some text.......
  </Popover>
}
```

**Comment by dimovpetar** [comment]
Hello @SAP/ui5-webcomponents-topic-core ,

As we discussed, can you check if the property `opener` can also accept objects, not only strings?

Best regards,
Petar

**Comment by ilhan007** [comment]
Hello @C3ntraX the issue has been addressed via this PR Addressed with:
https://github.com/SAP/ui5-webcomponents/pull/5609 which is available since 1.7.0
